addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -229,36 +229,39 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
             }
         }
         async _getProductByBarcode(code) {
-            let product = this.env.pos.db.get_product_by_barcode(code.base_code);
-            if (!product) {
-                // find the barcode in the backend
-                let foundProductIds = [];
-                try {
-                    foundProductIds = await this.rpc({
-                        model: 'product.product',
-                        method: 'search',
-                        args: [[['barcode', '=', code.base_code], ['sale_ok', '=', true]]],
-                        context: this.env.session.user_context,
-                    });
-                } catch (error) {
-                    if (isConnectionError(error)) {
-                        return this.showPopup('OfflineErrorPopup', {
-                            title: this.env._t('Network Error'),
-                            body: this.env._t("Product is not loaded. Tried loading the product from the server but there is a network error."),
-                        });
-                    } else {
-                        throw error;
-                    }
+            const barcode = code.base_code;
+            const product = this.env.pos.db.get_product_by_barcode(barcode) 
+                             || await this._findProduct(barcode) 
+                             || this._barcodeErrorAction(code);
+            return product;
+        }
+        async _findProduct(barcode) {
+            // find the barcode in the backend
+            let product;
+            try {
+                let foundProducts = await this.rpc({
+                    model: 'pos.session',
+                    method: 'find_product_by_barcode',
+                    args: [odoo.pos_session_id, barcode],
+                    context: this.env.session.user_context,
+                });
+                if (foundProducts) {
+                    const { product: foundProduct, packaging: foundPackaging = [] } = foundProducts;
+                    await this.env.pos._loadProductProduct(foundProduct);
+                    this.env.pos.db.add_packagings(foundPackaging);
+                    product = this.env.pos.db.get_product_by_id(foundProduct[0].id);
```

Comment:
can you confirm that when the product packaging is properly added thru add_packagings, the scanning of packaging is handled properly in `_barcodeProductAction`?